builders/bionic/builder.toml

```diff
@@ -1,4 +1,14 @@
 # Buildpacks to include in builder
+[[buildpacks]]
+id = "samples/curl"
+version = "0.0.1"
+uri = "../../extensions/curl"
```

Comment:
If the extensions uri is part of the `[[buildpacks]]` block, how the `builder` tool will be able to package within the image, the buildpacks containing the `bin/detect, bin/build` vs extensions as the existing uri parameter is used to fetch buildpacks content only ?

Why don't we package the extensions part of a separate block `[[extensions]]`.
Why such a proposition: 
- To be able to resolve the problem reported in my previous remark when the builder tool is creating an image,
- To allow to add new extensions when new builder images are created independently of the existing buildpacks
- To make the extension independent of the buildpacks*

*: If the buildpack detects that a needed tool, package is needed (curl, maven, java, ...), then it can report (`maven 3.x is needed), then the `extension` phase-step could then be called to install curl, java, ... for the suggested version before the `build` phase is called

Reply:
Submitted a PR for this change: https://github.com/buildpacks/samples/pull/115.

Reply:
That is an interesting question about the order. My understanding is from the RFC is that extensions should always come first in the order. The separate `[[order-ext]]` table is a bit confusing to me, as what would happen if multiple groups are defined there? This would change the lifecycle implementation (`detect` would happen separately for extensions vs. buildpacks).